### PR TITLE
[new release] js_of_ocaml, js_of_ocaml-ocamlbuild, js_of_ocaml-ppx_deriving_json, js_of_ocaml-tyxml, js_of_ocaml-ppx, js_of_ocaml-lwt, js_of_ocaml-compiler and js_of_ocaml-toplevel (3.5.0)

### PR DESCRIPTION
**Pull request by hhugo**
Compiler from OCaml bytecode to Javascript

- Project page: <a href="http://ocsigen.github.io/js_of_ocaml">http://ocsigen.github.io/js_of_ocaml</a>

## Features/Changes
* Compiler: Improve testing of the compiler (Ty Overby)
* Compiler: Add several macros for making the runtime easier to maintain (ocsigen/js_of_ocaml#771) (Ty Overby)
* Compiler: Allow to emit one javascript per compilation unit (ocsigen/js_of_ocaml#783)
* Compiler: refactoring (ocsigen/js_of_ocaml#781, ocsigen/js_of_ocaml#782, ocsigen/js_of_ocaml#787, ocsigen/js_of_ocaml#795, ocsigen/js_of_ocaml#802)
* Compiler: more source map location for the javascript runtime (ocsigen/js_of_ocaml#795)
* Compiler: tune variable naming (ocsigen/js_of_ocaml#838)
* Compiler: Work around num lib incompatibility
* Compiler: escape '</' in strings (ocsigen/js_of_ocaml#899)
* Compiler: speedup toplevel creation
* Runtime: support sharing when marshaling (ocsigen/js_of_ocaml#814)
* Runtime: add caml_obj_with_tag
* Runtime: support marshaling custom block
* Runtime: complete rewrite of bigarray
* Runtime: complete num implementation
* Runtime: add caml_ba_hash
* Runtime: rewrite polymorphic compare
* Ppx: switch ppx rewriter to the OCaml 4.08 ast
* Misc: Improve CI speed
* Misc: remove ppx_deriving dependency
* Misc: remove cppo dependency
* Misc: remove ppx_tools_versioned dependency in ppx_deriving_json
* Misc: support for ocaml 4.09
* Misc: switch to ocamlformat.0.12
* Misc: many more tests
* Misc: new jsoo_fs tool to embed files in a jsoo pseudo fs.
* Lib: Use expect tests
* Lib: Add support for 'addEventListener' with options (ocsigen/js_of_ocaml#807)
* Lib: Change api of [Lwt_js_events.async] (ocsigen/js_of_ocaml#862)
* Lib: Change api of responseText in xmlhttprequest (ocsigen/js_of_ocaml#863)
* Lib: add resizeObserver bindings
* Lib: Added support for custom events (ocsigen/js_of_ocaml#877)
* Lib: Added support for focus events (ocsigen/js_of_ocaml#885)
* Lib: Added `passive` option support for `Lwt_js_events` module
* Lib: Added bindings for pointer events (ocsigen/js_of_ocaml#894)

## Bug fixes
* Compiler: don't generate source if no-source-map passed (ocsigen/js_of_ocaml#780)
* Compiler: Fix compilation of [Array.set] to return [unit]/0 (ocsigen/js_of_ocaml#792)
* Compiler: Fix assertion failure (ocsigen/js_of_ocaml#828)
* Compiler: Fix compilation of exception handlers (ocsigen/js_of_ocaml#830)
* Compiler: Fix static evaluation of caml_equal (ocsigen/js_of_ocaml#906)
* Misc: Fix install on windows (ocsigen/js_of_ocaml#794)
* Lib: Fix Dom_svg.createForeignObject (ocsigen/js_of_ocaml#756)
* Runtime: Fix caml_obj_tag, causing miscompilation with lazy value (ocsigen/js_of_ocaml#772)
* Runtime: Fix caml_ml_seek_out, caml_ml_pos_out (ocsigen/js_of_ocaml#779) (Shachar Itzhaky)
* Runtime: caml_parse_sign_and_base to support unsigned syntax (ocsigen/js_of_ocaml#792) (Shachar Itzhaky)
* Runtime: fix encoding when printing to stdout (ocsigen/js_of_ocaml#800)
* Runtime: Handle browserfs in fs_node detection logic (ocsigen/js_of_ocaml#831)
* Runtime: fix Obj.tag (ocsigen/js_of_ocaml#832)
* Runtime: fix marshalling of custom blocks (ocsigen/js_of_ocaml#861)
* Runtime: fix frexp
* Runtime: fix float printing with "%f" and large floats

### CODE REVIEW

File: packages/js_of_ocaml-tyxml/js_of_ocaml-tyxml.3.5.0/opam [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "ocaml" {>= "4.02.0"}
+  "dune" {build & >= "1.11.1"}
+  "tyxml" {>= "4.3"}
+  "reactiveData" {>= "0.2"}
+  "js_of_ocaml" {>= "3.0"}
+  "js_of_ocaml-ppx"
+]
+conflicts: [
+  "js_of_ocaml"             {<"3.0"}
+]
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-tyxml/js_of_ocaml-tyxml.3.5.0/opam]: This field is redundant with the constraints already present in the `depends` field.

File: packages/js_of_ocaml-compiler/js_of_ocaml-compiler.3.5.0/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Js_of_ocaml is a compiler from OCaml bytecode to JavaScript.
+It makes it possible to run pure OCaml programs in JavaScript
+environment like browsers and Node.js
+"""
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "ocaml" {>= "4.02.0"}
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-compiler/js_of_ocaml-compiler.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

File: packages/js_of_ocaml-lwt/js_of_ocaml-lwt.3.5.0/opam [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+Js_of_ocaml is a compiler from OCaml bytecode to JavaScript.
+It makes it possible to run pure OCaml programs in JavaScript
+environment like browsers and Node.js
+"""
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "ocaml" {>= "4.02.0"}
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-lwt/js_of_ocaml-lwt.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

File: packages/js_of_ocaml-ocamlbuild/js_of_ocaml-ocamlbuild.3.5.0/opam [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+description: """
+Js_of_ocaml is a compiler from OCaml bytecode to JavaScript.
+It makes it possible to run pure OCaml programs in JavaScript
+environment like browsers and Node.js
+"""
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-ocamlbuild/js_of_ocaml-ocamlbuild.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

File: packages/js_of_ocaml-ppx/js_of_ocaml-ppx.3.5.0/opam [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+Js_of_ocaml is a compiler from OCaml bytecode to JavaScript.
+It makes it possible to run pure OCaml programs in JavaScript
+environment like browsers and Node.js
+"""
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "ocaml" {>= "4.02.0"}
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-ppx/js_of_ocaml-ppx.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

File: packages/js_of_ocaml-ppx_deriving_json/js_of_ocaml-ppx_deriving_json.3.5.0/opam [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+Js_of_ocaml is a compiler from OCaml bytecode to JavaScript.
+It makes it possible to run pure OCaml programs in JavaScript
+environment like browsers and Node.js
+"""
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "ocaml" {>= "4.04.1"}
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-ppx_deriving_json/js_of_ocaml-ppx_deriving_json.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

File: packages/js_of_ocaml-toplevel/js_of_ocaml-toplevel.3.5.0/opam [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+Js_of_ocaml is a compiler from OCaml bytecode to JavaScript.
+It makes it possible to run pure OCaml programs in JavaScript
+environment like browsers and Node.js
+"""
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "ocaml" {>= "4.02.0"}
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-toplevel/js_of_ocaml-toplevel.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

File: packages/js_of_ocaml-tyxml/js_of_ocaml-tyxml.3.5.0/opam [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+Js_of_ocaml is a compiler from OCaml bytecode to JavaScript.
+It makes it possible to run pure OCaml programs in JavaScript
+environment like browsers and Node.js
+"""
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "ocaml" {>= "4.02.0"}
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml-tyxml/js_of_ocaml-tyxml.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

File: packages/js_of_ocaml/js_of_ocaml.3.5.0/opam [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+environment like browsers and Node.js
+"""
+
+name: "js_of_ocaml"
+
+build: [["dune" "build" "-p" name "-j" jobs]]
+
+depends: [
+  "ocaml" {>= "4.02.0"}
+  "dune" {build & >= "1.11.1"}
```
Comment (kit-ty-kate) [inline packages/js_of_ocaml/js_of_ocaml.3.5.0/opam]:
```suggestion
  "dune" {>= "1.11.1"}
```
Dune should not be tagged as a build dependency (see #14266)

**Comment by hhugo** [comment]
the last error is a timeout with macOS

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**